### PR TITLE
Translations: Convert JSON to TypeScript

### PR DESCRIPTION
**Pull request by AnnikaCodes**
I also made `language` attributes IDs since that's already how they were being used.

### CODE REVIEW

File: translations/traditionalchinese/minor-activities.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import {Translations} from "../../server/chat";
```
Comment (Zarel) [inline translations/traditionalchinese/minor-activities.ts:1]:
```suggestion
import type {Translations} from "../../server/chat";
```

(and so on for all the others)

File: server/chat.ts [hunk 2]
```diff
@@ -97,6 +97,15 @@ export type PunishmentFilter = (user: User | ID, punishment: Punishment) => void
...
 export type HostFilter = (host: string, user: User, connection: Connection, hostType: string) => void;
 
+export interface TRStrings {
+	[k: string]: string;
+}
+
+export interface Translations {
+	name?: string;
+	strings: TRStrings;
+}
```
Comment (Zarel) [inline server/chat.ts:107]:
```suggestion
export interface Translations {
	name?: string;
	strings: {[english: string]: string};
}
```

It doesn't look like you actually use the `TRStrings` type anywhere.

Comment (AnnikaCodes) [reply server/chat.ts:107]: Hm, I think that was just a leftover from before.

File: .eslintrc-no-types.json [hunk 3]
```diff
@@ -192,7 +192,7 @@
     {
       "files": [
         "./config/*.ts", "./data/**/*.ts", "./lib/*.ts", "./server/**/*.ts", "./sim/**/*.ts",
-        "./tools/set-import/*.ts",  "./tools/modlog/*.ts"
+         "./tools/set-import/*.ts",  "./tools/modlog/*.ts", "./translations/**/*.ts"
```
Comment (Zarel) [inline .eslintrc-no-types.json:195]:
```suggestion
        "./tools/set-import/*.ts",  "./tools/modlog/*.ts", "./translations/**/*.ts"
```

**Comment by Zarel** [comment]
Thanks!

**Pull request merged**